Sample UI improvements for iPhone X

nCastle1 opened the pull request:
PR is WIP

fragmer commented on src/iOS/Xamarin.iOS/Samples/Data/StatsQueryGroupAndSort/StatsQueryGroupAndSort.cs:
```diff
@@ -748,40 +728,20 @@ public class ChooseStatisticOverlay : UIView
 
             // Add Button (add the new stat and don't dismiss the UI).
-            UIBarButtonItem addButton = new UIBarButtonItem("Add", UIBarButtonItemStyle.Done, (s, e) =>
+            UIBarButtonItem addButton = new UIBarButtonItem("Add", UIBarButtonItemStyle.Plain, (s, e) =>
```
Almost forgot one more:
Clicking "Add" button on "Stat definitions" screen does nothing unless user changes something in the picker first.

nCastle1 commented:
@dkemlage Thanks for the review! I've fixed the two crashes you found, which were indeed a side effect of the changes I made. 

@fragmer Thanks for the review! I've fixed all but the last two comments. There's some trickiness with keyboard types that I'd need to account for (in particular, the lack of a return key). Combined with the complexity of implementing given the current structure, it might be better to wait until the sample is redesigned. For stats query, there would need to be significant restructuring to avoid the issue you ran into.
